components/purchase_modal/purchase_modal.tsx

```diff
@@ -192,7 +192,9 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
     listPlans = () => {
         const products = this.props.products!;
         const options = Object.keys(products).map((key: string) => {
-            return {key: products[key].name, value: products[key].id};
+            return {key: products[key].name, value: products[key].id, price: products[key].price_per_seat};
+        }).sort((a, b) => a.price - b.price).filter((option) => {
+            return option.price >= this.state.currentProduct!.price_per_seat;
```

Comment:
@AGMETEOR this code has now changed to consider the `flat_fee` subscriptions in this PR: https://github.com/mattermost/mattermost-webapp/pull/8341/files#diff-acb502910cc86e76a73b356eba7ef5024d0db3aaa31441bbe70ff3c83dc164beR193-R206

We can now update this PR and consider not allowing them to downgrade but *ONLY* when you are upgrading (not the initial purchase from trial given that they may be in Enterprise and can select Starter to subscribe)

Reply:
I have updated the code